src/backings/readOnlyIterate.ts

```diff
@@ -0,0 +1,25 @@
+import {ArrayLike} from "../interface";
+
+export function readOnlyForEach<T>(value: ArrayLike<T>, fn: (value: T, index: number) => void): void {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  if ((value as any).readOnlyForEach) {
```

Comment:
Can you create ReadOnlyIterator<T> type?
```typescript
ReadOnlyIterator<T> {
    readOnlyForEach: (value: T, index: number) => void;
    readOnlyMap<U>: (value: T, index: number) => U:
}

```


You would have `value: ArrayLike<T>|ReadOnlyIterator<T>` without any casting